Makefile

```diff
@@ -21,6 +24,13 @@ build:
 	$(GOENV) $(GOBUILD) -o ./out/$(ARCHIVE_NAME)/cacher -a ./cmd/cacher
 	$(GOENV) $(GOBUILD) -o ./out/$(ARCHIVE_NAME)/launcher -a ./cmd/launcher
 
+descriptor: install-yj
+	@echo "> Writing descriptor file..."
+	mkdir -p ./out
+	echo '{"api": {"platform": "$(PLATFORM_API)", "buildpack": "$(BUILDPACK_API)"}, "lifecycle": {"version": "$(LIFECYCLE_VERSION)"}}' \
```

Comment:
Is there a reason why we did just right a toml file from a multiline string?

```bash
cat >./out/lifecycle.toml <<EOL
[api]
 platform = "$(PLATFORM_API)"
 buildpack = "$(BUILDPACK_API)"

[lifecycle]
  version = "$(LIFECYCLE_VERSION)"
EOL
```

This would achieve two things...

1. Remove dependency on `yj`
2. Readability

Reply:
I was under the impression `Makefile`s don't support HEREDOCs

Reply:
You are right... 

After some sweat and tears, I did find a decent alternative for `Makefile` based on [this stackoverflow answer](https://stackoverflow.com/a/10960333).


```makefile
LIFECYCLE_VERSION?=0.0.0
PLATFORM_API=0.1
BUILDPACK_API=0.2
define LIFECYCLE_TOML_BODY
[api]
  platform = "$(PLATFORM_API)"
  buildpack = "$(BUILDPACK_API)"

[lifecycle]
  version = "$(LIFECYCLE_VERSION)"
endef

descriptor: export LIFECYCLE_TOML_BODY:=$(LIFECYCLE_TOML_BODY)
descriptor:
	mkdir -p out
	echo "$${LIFECYCLE_TOML_BODY}" > ./out/lifecycle.toml
```